File: docs/scripts/endOfGame/HighestScore.js
import CellState from "../CellState.js";
import Player from "../Player.js";
import Winner from "../Winner.js";

export default class HighestScore {
    constructor(c) {
        this.game = c;
    }
    condition() {
        for(let cs of Object.keys(CellState).slice(0, -1)) {
            if (this.game.canPlay(cs)) {
                return Winner.NONE;
            }    
        }
        let p1 = this.game.scores[Player.PLAYER1];
        let p2 = this.game.scores[Player.PLAYER2];
        if (p1 > p2) {
            return Winner.PLAYER1;
        }
        if (p1 < p2) {
            return Winner.PLAYER2;
        }
        if (p1 === p2) {
            return Winner.DRAW;
        }
    }
}

File: src/main/java/model/endOfGame/HighestScore.java
package model.endOfGame;

import java.util.Arrays;

import model.CellState;
import model.LeapFrog;
import model.Player;
import model.Winner;

public class HighestScore implements EndOfGameStrategy {
    private LeapFrog game;

    public HighestScore(LeapFrog c) {
        this.game = c;
    }

    public Winner condition() {
        CellState[] temp = Arrays.copyOf(CellState.values(), CellState.values().length - 1);
        for (CellState cs : temp) {
            if (this.game.canPlay(cs)) {
                return Winner.NONE;
            }
        }
        int p1 = this.game.getScores().get(Player.PLAYER1);
        int p2 = this.game.getScores().get(Player.PLAYER2);
        if (p1 > p2) {
            return Winner.PLAYER1;
        }
        if (p1 < p2) {
            return Winner.PLAYER2;
        }
        return Winner.DRAW;
    }
}
